docs/contributor/arch/017-otel-log-agent-fault-toleration.md

```diff
@@ -0,0 +1,110 @@
+# 17. OTel Log Agent Fault Toleration 
+
+Date: 2025-02-04
+
+## Status
+
+Proposed
+
+## Context
+
+### Classifying Log Loss Scenarios in an OTel Log Agent with Filelog Receiver
+When setting up an OpenTelemetry (OTel) Log Agent using the Filelog Receiver, it's important to identify situations where logs might be lost and how to mitigate them.
+
+#### Scenarios Where Data Loss Must Be Prevented:
+- Temporary OTLP backend issues (e.g., spikes in retriable errors, backpressure, temporary network failures).
+- Collector Pod restarts during normal operations (e.g., upgrades, rescheduling to another node).
+
+#### Scenarios Where Preventing Data Loss Is Nice-to-Have:
+- Collector Pod crashes that occur unexpectedly.
+
+#### Scenarios Where Data Loss Is Unavoidable:
+- Permanent OTLP backend failures.
+- Node-level failures.
+- Logs not yet tailed before Collector Pod eviction.
+- Logs not yet tailed before being rotated.
+
+### Mechanisms for Enhanced Resiliency  
+#### Batch Processor
+The Batch Processor accepts logs and places them into batches. Batching helps better compress the data and reduce the number of outgoing connections required to transmit the data. However, there are some problems:
```

Comment:
is it really the number of connections of just the number of requests?

Reply:
Number of requests, you are right